datafusion/sqllogictest/test_files/union.slt

```diff
@@ -763,9 +763,5 @@ SELECT NULL WHERE FALSE;
 1
 
 # Test Union of List Types. Issue: https://github.com/apache/datafusion/issues/12291
-query error
-SELECT make_array(2) x UNION ALL SELECT make_array(now()) x;
-----
-DataFusion error: type_coercion
-caused by
-Error during planning: Incompatible inputs for Union: Previous inputs were of type List(Field { name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), but got incompatible type List(Field { name: "item", data_type: Timestamp(Nanosecond, Some("+00:00")), nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }) on column 'x'
+query error DataFusion error: type_coercion
```

Comment:
This works, but doesn't prove that the error message is useful and actionable when end user sees it.
Let's include more detail. It's ok to keep full error message, or write a pattern to make it shorter.

Reply:
fair! Expanded the error in https://github.com/apache/datafusion/pull/13452/commits/d169abcedf979602cbd689a332a576ffcc2ad4bf